crates/ruff_linter/src/rules/flake8_pytest_style/rules/helpers.rs

```diff
@@ -45,6 +48,66 @@ pub(super) fn is_pytest_parametrize(call: &ExprCall, semantic: &SemanticModel) -
             matches!(qualified_name.segments(), ["pytest", "mark", "parametrize"])
         })
 }
+/// Whether `path` likely leads to a Pytest test file.
+///
+/// Pytest determines a test file using [a number of conventions and settings][1].
+/// Only the conventions are reflected here.
+/// This function returns true if the file name (`.py` notwithstanding):
+///
+/// * Starts with `test_`, or
+/// * Ends with `_test`
+///
+/// It is thus prone to both false positives and false negatives.
+///
+/// [1]: https://docs.pytest.org/en/stable/explanation/goodpractices.html#conventions-for-python-test-discovery
+#[allow(dead_code)]
+pub(crate) fn is_likely_pytest_test_file(path: &Path) -> bool {
+    if !matches!(path.extension().and_then(OsStr::to_str), Some("py")) {
+        return false;
+    }
+
+    let Some(file_stem) = path.file_stem().and_then(OsStr::to_str) else {
+        return false;
+    };
+
+    file_stem.starts_with("test_") || file_stem.ends_with("_test")
+}
+
+/// Whether the currently checked `func` is likely to be a Pytest test.
+///
+/// A normal Pytest test function is one whose name starts with `test` and is either:
+///
+/// * Placed at module-level, or
+/// * Placed within a class whose name starts with `Test` and does not have an `__init__` method.
+///
+/// See also [`is_likely_pytest_test_file`].
+pub(crate) fn is_likely_pytest_test(func: &StmtFunctionDef, checker: &Checker) -> bool {
```

Comment:
Are there cases where this method returns `false` for a proper `pytest`? If not, I suggest renaming it ti `is_pytest_test`. 

We should document whether the function has false negatives or false positives and, if so, when they occur.

Let's also include this a link to https://docs.pytest.org/en/stable/explanation/goodpractices.html#conventions-for-python-test-discovery . I found it a very useful reference to have